File: tests/codebase/fixtures/sample.py
"""Fixture for T5 — Python extractor."""
import os
from collections import deque

def helper(x):
    return x + 1

class Greeter:
    def greet(self):
        return helper(self)

def outer():
    def inner_helper():
        return 1

def outer():  # type: ignore[no-redef]
    def inner_helper():
        return 2

helper(1); helper(2)

File: tests/codebase/fixtures/sample.ts
import { foo } from "./bar";
import * as ns from "lib";

function helper(x: number): number { return x + 1; }

class Greeter {
  greet(name: string): string { return helper(1); }
}

interface Animal { name: string; }
const arrow = (x: number) => helper(x);

function outer() { function inner_helper() { return 1; } }
function outer() { function inner_helper() { return 2; } }

helper(1); helper(2);
